drivers/modem/gsm_ppp.c

```diff
@@ -253,13 +267,43 @@ static const struct setup_cmd setup_cmds[] = {
 	SETUP_CMD("AT+CGSN", "", on_cmd_atcmdinfo_imei, 0U, ""),
 #endif
 
-	/* disable unsolicited network registration codes */
-	SETUP_CMD_NOHANDLE("AT+CREG=0"),
+	/* enable network registration report */
+	SETUP_CMD_NOHANDLE("AT+CREG=1"),
```

Comment:
I guess there's no sense to enable NRCs as polling is done manually?

Reply:
I think you are right, initially I enabled this to see if the modem would send the registration report periodically so that I can have a callback to my application but I think my Quectel modem doesn't really send the report unless I perform a manual query.

Will remove it when I'm back to this PR branch